mmv1/products/databasemigrationservice/MigrationJob.yaml

```diff
@@ -0,0 +1,276 @@
+      - !ruby/object:Api::Type::String
+        name: 'message'
+        output: true
+        description: |
+          Human readable message indicating details about the current status.
+      - !ruby/object:Api::Type::Array
+        name: 'details'
+        description: |
+          A list of messages that carry the error details.
+        item_type: Api::Type::KeyValuePairs
+        output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'type'
+    description: |
+      The type of the migration job.
+    values:
+      - :ONE_TIME
+      - :CONTINUOUS
+  - !ruby/object:Api::Type::String
+    name: 'source'
+    description: |
+      The name of the source connection profile resource in the form of projects/{project}/locations/{location}/connectionProfiles/{sourceConnectionProfile}.
+  - !ruby/object:Api::Type::String
+    name: 'destination'
+    description: |
+      The name of the destination connection profile resource in the form of projects/{project}/locations/{location}/connectionProfiles/{destinationConnectionProfile}.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'dumpFlags'
+    description: |
+      The initial dump flags.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: 'dumpFlags'
+        description: |
+          A list of dump flags
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'name'
+              description: |
+                The name of the flag
+            - !ruby/object:Api::Type::String
+              name: 'value'
+              description: |
+                The vale of the flag
+  - !ruby/object:Api::Type::NestedObject
+    name: 'performanceConfig'
+    description: |
+      Data dump parallelism settings used by the migration.
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'dumpParallelLevel'
+        description: |
+          Initial dump parallelism level.
+        values:
+          - :MIN
+          - :OPTIMAL
+          - :MAX
+  - !ruby/object:Api::Type::String
+    name: 'dumpPath'
+    description: |
+      The path to the dump file in Google Cloud Storage,
+      in the format: (gs://[BUCKET_NAME]/[OBJECT_NAME]).
+      This field and the "dump_flags" field are mutually exclusive.
+  - !ruby/object:Api::Type::Enum
+    name: 'dumpType'
+    description: |
+      The type of the data dump. Supported for MySQL to CloudSQL for MySQL
+      migrations only.
+    values:
+      - :LOGICAL
+      - :PHYSICAL
+  - !ruby/object:Api::Type::NestedObject
+    name: 'staticIpConnectivity'
+    description: |
+      static ip connectivity data (default, no additional details needed).
+    exactly_one_of:
+      - reverseSshConnectivity
+      - vpcPeeringConnectivity
+    allow_empty_object: true
+    send_empty_value: true
+    properties: []
+  - !ruby/object:Api::Type::NestedObject
```

Comment:
This object+below should any of the subfields be required?

Reply:
Marked required fields as such.